client/api/analytics/AnalyticsManager.cs

```diff
@@ -54,16 +54,14 @@ public void Stop()
         public void PushToCache(Target target, FeatureConfig featureConfig, Variation variation)
         {
             var cacheSize = analyticsCache.GetAllElements().Count;
-            var bufferSize = config.getBufferSize();
+            var bufferSize = config.bufferSize;
 
             if (cacheSize > bufferSize)
```

Comment:
Thinking about it again this logic still isn't perfect. 
It might be sufficient for now, but it probably doesn't achieve what we want.    We care about unique keys i.e. flag,variation,target.     If the current evaluation matches an existing key we only want to increment the count for that flag.    If it doesn't match and the cacheSize > buffer then we want to discard it.

Should we actually push this logic further down into https://github.com/harness/ff-dotnet-server-sdk/blob/da071ea1b2fa54e2e7e0a98111e4dc414036e636/client/api/analytics/AnalyticsManager.cs#L90